transport/src/main/java/io/netty/channel/DefaultChannelHandlerContext.java

```diff
@@ -817,8 +803,10 @@ public ChannelFuture writeAndFlush(Object msg) {
         return writeAndFlush(msg, newPromise());
     }
 
-    private static void notifyOutboundHandlerException(Throwable cause, ChannelPromise promise) {
-        PromiseNotificationUtil.tryFailure(promise, cause, logger);
+    private void handleOutboundHandlerException(Throwable cause) {
```

Comment:
What do you think about removing `fireExceptionCaught()` and `exceptionCaught()` all together and have  `close()` (both for `ChannelHandler` and `ChannelHandlerContext`) take an argument of `Throwable` to indicate if the channel is closed due to an error.

Reply:
I think this is not a good idea as there might be situations where a exception can be recovered etc.

Reply:
If we are unsure then perhaps a good idea to list those situations. I am generally doubtful that an exception generated by one component is recoverable by another using a callback (event on the pipeline)

Reply:
Even if we want to do what you suggest I think we should make this another PR and discuss there. That said I am not convinced it is a good idea at all.